src/components/PressableWithSecondaryInteraction/index.native.tsx

```diff
@@ -0,0 +1,57 @@
+import React, {forwardRef} from 'react';
+import {GestureResponderEvent, TextProps} from 'react-native';
+import {PressableRef} from '@components/Pressable/GenericPressable/types';
+import PressableWithFeedback from '@components/Pressable/PressableWithFeedback';
+import Text from '@components/Text';
+import PressableWithSecondaryInteractionProps from './types';
+
+/** This is a special Pressable that calls onSecondaryInteraction when LongPressed. */
+function PressableWithSecondaryInteraction(
+    {
+        children,
+        onSecondaryInteraction,
+        inline = false,
+        needsOffscreenAlphaCompositing = false,
+        activeOpacity = 1,
+        preventDefaultContextMenu,
+        withoutFocusOnSecondaryInteraction,
+        enableLongPressWithHover,
+        ...rest
+    }: PressableWithSecondaryInteractionProps,
+    ref: PressableRef,
+) {
+    const executeSecondaryInteraction = (event: GestureResponderEvent) => {
+        event.preventDefault();
+        onSecondaryInteraction?.(event);
+    };
+
+    // Use Text node for inline mode to prevent content overflow.
+    if (inline) {
+        return (
+            <Text
+                // eslint-disable-next-line react/jsx-props-no-spreading
+                {...(rest as TextProps)}
+                onLongPress={onSecondaryInteraction ? executeSecondaryInteraction : undefined}
+            >
```

Comment:
I checked, and `inline` is set to `true` only one time and ref isn't used. The other thing is how to type `ref`, I find it very difficult when typing it `ForwardedRef<Text | View>`, any ideas? @BartoszGrajdek @kubabutkiewicz

Reply:
First thing that came to my mind was to use overload, however all these types here are so intertwined that I'm not sure if it would even work and if it would be a valid solution.